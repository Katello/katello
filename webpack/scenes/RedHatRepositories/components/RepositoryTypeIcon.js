import React from 'react';
import PropTypes from 'prop-types';
import { ListView, OverlayTrigger, Tooltip } from 'patternfly-react';

import { getTypeIcon } from '../../../services/index';

export default class RepositoryTypeIcon extends React.Component {
  constructor(props) {
    super(props);

    this.tooltipId = `type-tooltip-${props.id}`;
  }

  render() {
    return (
      <OverlayTrigger
        overlay={<Tooltip id={this.tooltipId}>{this.props.type}</Tooltip>}
        placement="bottom"
        trigger={['hover', 'focus']}
        rootClose={false}
      >
        <ListView.Icon name={getTypeIcon(this.props.type)} />
      </OverlayTrigger>
    );
  }
}

RepositoryTypeIcon.propTypes = {
  id: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
};
