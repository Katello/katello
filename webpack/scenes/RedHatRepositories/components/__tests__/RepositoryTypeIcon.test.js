import React from '@theforeman/vendor/react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import RepositoryTypeIcon from '../RepositoryTypeIcon';

describe('RepositoryTypeIcon component', () => {
  const getBaseProps = () => ({
    id: 123,
    type: 'unknown-type',
  });

  describe('rendering', () => {
    test('it should get rendered correctly', () => {
      const component = mount(<RepositoryTypeIcon {...getBaseProps()} />);

      expect(toJson(component)).toMatchSnapshot();
    });

    test('it should get rendered correctly when type is provided', () => {
      const props = getBaseProps();
      props.type = 'yum';
      const component = mount(<RepositoryTypeIcon {...props} />);

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
