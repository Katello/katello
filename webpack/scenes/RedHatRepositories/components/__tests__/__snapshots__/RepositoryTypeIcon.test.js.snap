// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryTypeIcon component rendering it should get rendered correctly 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Tooltip
      bsClass="tooltip"
      id="type-tooltip-123"
      placement="right"
    >
      unknown-type
    </Tooltip>
  }
  placement="bottom"
  rootClose={false}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <ListViewIcon
    name="fa-2x fa fa-question"
  />
</OverlayTrigger>
`;

exports[`RepositoryTypeIcon component rendering it should get rendered correctly when type is provided 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Tooltip
      bsClass="tooltip"
      id="type-tooltip-123"
      placement="right"
    >
      yum
    </Tooltip>
  }
  placement="bottom"
  rootClose={false}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <ListViewIcon
    name="fa-2x pficon-bundle"
  />
</OverlayTrigger>
`;
